src/environment.cpp

```diff
@@ -175,21 +175,23 @@ void Environment::continueRaycast(RaycastState *state, PointedThing *result)
 			bool is_colliding = false;
 			// Minimal distance of all collisions
 			float min_distance_sq = 10000000;
+			// ID of the current box (loop counter)
+			u16 id = 0;
 
 			v3f npf = intToFloat(np, BS);
-			for (std::vector<aabb3f>::const_iterator i = boxes.begin();
-					i != boxes.end(); ++i) {
-				// Get current collision box
-				aabb3f box = *i;
+			// This loop translates the boxes to their in-world place.
+			for (aabb3f &box : boxes) {
```

Comment:
@nerzhul Is this still a problem? `boxes` is not used after this loop, and using its elements by reference spares a memory copy.

Reply:
I think this is fine